build/run-dev-env-in-docker.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a script that runs gulp in a docker container,
+# for machines that don't have nodejs, go and java installed.
+
+DOCKER_RUN_OPTS=${DOCKER_RUN_OPTS:-}
+DASHBOARD_IMAGE_NAME="kubernetes-dashboard-build-image"
+CD="$(pwd)"
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+K8S_DASHBOARD_SRC=${K8S_DASHBOARD_SRC:-"${CD}"}
+
+K8S_DASHBOARD_CONTAINER_NAME=${K8S_DASHBOARD_CONTAINER_NAME:-"k8s-dashboard-dev"}
+
+echo "Remove existing container"
+docker rm -f ${K8S_DASHBOARD_CONTAINER_NAME}
+
+# Always test if the image is up-to-date. If nothing has changed since last build,
+# it'll just use the already-built image
+docker build -t ${DASHBOARD_IMAGE_NAME} -f ${DIR}/Dockerfile ${DIR}/../
+
+# Run dashboard container for development and expose necessary ports automatically.
+docker run \
```

Comment:
Already the script to run a process is there, i.e. build/run-gulp-in-docker.sh, that uses same Dockerfile. But if we change source files, the "docker build" can not use cache for tasks after copying sources. i.e. for installing dependencies, and it takes a while in each execution of a process.

I don't know who uses existing script, so I proposed adding new script as now. If nobody uses the existing script, especially any other scripts for test or deploy does not use it, it may be better to improve the exisiting one.